server/scripts/populate.py

```diff
@@ -0,0 +1,180 @@
+    community = "Community"
+    tech = "Tech"
+    creative = "Creative"
+    active = "Active"
+    volunteering = "Volunteering"
+    gaming = "Gaming"
+    career = "Career"
+    engineering = "Engineering"
+    science = "Science"
+    environment = "Environment"
+    arts = "Arts"
+    math = "Math"
+    health = "Health"
+
+
+class EventSchema(Schema):
+    name = fields.Str(required=True, validate=validate.Length(min=1))
+    description = fields.Str(required=True, validate=validate.Length(min=1))
+    url = fields.Url()
+    banner_image = fields.Url(required=True, validate=validate.Length(min=1))
+    start_time = fields.DateTime(required=True)
+    end_time = fields.DateTime(required=True)
+    facebook = fields.Url()
+    twitter = fields.Url()
+    instagram = fields.Url()
+    tags = fields.List(fields.Str(
+        validate=validate.OneOf([tag.value for tag in Tags])))
+
+
+class ClubSchema(Schema):
+    name = fields.Str(required=True, validate=validate.Length(min=1))
+    description = fields.Str(required=True, validate=validate.Length(min=1))
+    size = fields.Int(required=True, validate=validate.Range(min=1))
+    banner_image = fields.Url(required=True, validate=validate.Length(min=1))
+    icon_image = fields.Url(required=True, validate=validate.Length(min=1))
+    facebook = fields.Url()
+    twitter = fields.Url()
+    instagram = fields.Url()
+    website = fields.Url()
+    events = fields.List(fields.Nested(EventSchema))
+    tags = fields.List(fields.Str(
+        validate=validate.OneOf([tag.value for tag in Tags])))
+
+
+def insert_tags():
+    tag_args = []
+    for tag in Tags:
+        _id = str(uuid.uuid4())
+        tag_entities[tag.value] = _id
+        tag_args.append((_id, datetime.now(), datetime.now(), tag.value))
+    with conn.cursor() as cursor:
+        cursor.execute("INSERT INTO tag VALUES" + ",".join(cursor.mogrify(
+            "(%s,%s,%s,%s)", x).decode("utf-8")
+            for x in tag_args))
+        conn.commit()
+    print("Tags Inserted")
+
+
+def read_clubs(filepath):
+    with open(filepath) as file:
+        club_dicts = yaml.load(file, Loader=yaml.FullLoader)
+    for club_dict in club_dicts:
+        club = ClubSchema().load(club_dict)
+        yield club
+
+
+
+def insert_clubs():
+    club_list = read_clubs(filepath)
+    if club_list is None:
+        print("Something failed; aborting...")
+        return
+
+    club_values = []
+    event_values = []
+    club_events_values = []
+    tag_events_values = []
+    tag_clubs_values = []
+
+    for club in club_list:
+        club["id"] = str(uuid.uuid4())
+        club_values.append((club.get("id"), datetime.now(), datetime.now(), club.get("name"),
+                            club.get("description"), club.get("size"), club.get(
+                                "banner_image"), club.get("icon_image"),
+                            club.get("facebook"), club.get("twitter"), club.get("instagram"), club.get("website")))
```

Comment:
likewise, lines like this can now be
```suggestion
                            club.facebook, club.twitter, club.instagram, club.website))
```

Reply:
I think .load() returns a dictionary by default. If we want an object, then there's an extra method that needs to be declared in a new Club class. See [here](https://marshmallow.readthedocs.io/en/stable/quickstart.html#deserializing-objects-loading). I'm not sure if Python's built-in dict->object is any more efficient than what I have here. Would like to hear what you think.

Reply:
There are tools like [`dataclasses-json`](https://pypi.org/project/dataclasses-json/) to cover the last step of converting to objects, but that would require redoing the schema and everything. What you have is fine for now